task6/script.js

```diff
@@ -0,0 +1,293 @@
+            createdAt: new Date('2021-03-17T23:00:00'),
+            author: 'Николай'
+        },
+
+    ];
+
+    
+    function getPost(id) {
+        for (let i = 0; i < posts.length ; i++) {
+            if (posts[i].id === id) {
+                return posts[i];
+            }
+        }
+    }
+    
+    function validatePost(post){
+        if(typeof post.id !== 'string' || post.id === ''){
+            return false;
+        }
+        if(typeof post.description !== 'string' || post.description === '' || post.description.length > 200)
+            return false;
+        if(!(post.createdAt instanceof Date)){
+            return false;
+        }
+        if(typeof post.author !== 'string' || post.author === ''){
+            return false;
+        }
+        return true;
+    }
+    
+    function addPost(post) {
+        if (validatePost(post)) {            
+            posts.push(post);
+            return true;
+        }
+        return false;
+    }
+    
+    function removePost(id){
+        for(let i = 0; i < posts.length; i++) {
+            if(posts[i].id === id){
+                posts.splice(i,1);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    
+    function editPost(id, post){
+        let editPost = getPost(id);
+        if(post.description !== '' && typeof post.description == 'string'){
+            editPost.description = post.description;
+        if(validatePost(editPost)){
+            for( let i = 0; i < posts.length; i++){
+                if(posts[i].id === id){
+                    posts[i] = editPost;
+                    return true;
+                }
+              }
+           }
+        }        
+        return false;
+    }
+    
+
+    function sortByDate(posts) {
+        posts.sort((post1,post2) => post1.createdAt - post2.createdAt);
+    }
+    
+    
+    function getPosts(skip = 0, top = 10, filterConfig) {
+        if(typeof skip !== 'number' || typeof top !== 'number'){
+            return 'некорректно заданы параметры';
+        }
+        let arr = [];
+        let searchArr = posts;
+        if (typeof filterConfig === 'undefined') {
+            arr = searchArr.slice(skip,skip+top);
+            sortByDate(arr);
+            return arr;
+        } 
+       
+            if (filterConfig.hasOwnProperty('author')) {
+                for(let i = 0; i < searchArr.length; i++){
+                    if(searchArr[i].author === filterConfig.author){
+                        arr.push(searchArr[i]);
+                    }                    
+                }
+                if(arr.length == 0){
+                   return 'нет такого автора';
+                }
+            }
+        
+        if(filterConfig.hasOwnProperty('from')){
+            for(let i = 0; i < searchArr.length; i++){
+                    if(Date.parse(searchArr[i].createdAt) >= Date.parse(filterConfig.from)){
+                        arr.push(searchArr[i]);
```

Comment:
Imagine, I have post from person `A` that was posted at time `3`. If I'll pass `{ author: 'A', from: 2, to: 4}` as `filterConfig`, it will return that post 3 times.
As an example, you can run `console.log(getPosts( 0,40, {author: 'Николай', to: '2023-03-25T23:00:00'}));` it return 22 posts, when there are only 20 exists